packages/fxa-auth-server/lib/metrics/glean/index.ts

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { ConfigType } from '../../../config';
+import { createAccountsEventsEvent } from './server_events';
+import { version } from '../../../package.json';
+import { createHash } from 'crypto';
+
+type MetricsData = {
+  uid?: string;
+};
+
+let appConfig: ConfigType;
+let gleanEventLogger: ReturnType<typeof createAccountsEventsEvent>;
+
+const isEnabled = async (request: any) =>
```

Comment:
Nit: narrow any type. This is in a few places below as well.

Reply:
Yeah, there's no existing request type for auth-server so I didn't use anything here.  Cannot simply use hapi's request type since the auth-server attaches numerous lazy getters to the value.  I'll see about declaring something for this file, although that feels a bit more letter of the law than spirit of the law.

Reply:
haha right after I posted that comment I remembered about the `AuthRequest` type.  I'll try using that.

Reply:
`AuthRequest` didn't _quite_ work right out of the tin.  I extended from it though.  I left some comments.